api/src/config.rs

```diff
@@ -11,10 +13,16 @@ use std::path::Path;
 use std::str::FromStr;
 use std::sync::atomic::{AtomicBool, Ordering};
 use std::sync::Arc;
+use std::sync::Mutex;
+use tokio;
 
 use serde::Deserialize;
 use serde_json::Value;
 
+lazy_static! {
+    pub static ref BACKEND_SOURCE_PATH: Mutex<Option<String>> = Mutex::new(None);
```

Comment:
How about putting this into `BackendConfigV2` struct like this (as a private field, do not participate in Deserialize/Serialize):

```
pub struct BackendConfigV2 {
    /// Type of storage backend.
    #[serde(rename = "type")]
    pub backend_type: String,
    /// Source of storage backend configuration.
    #[serde(skip_serializing, skip_deserializing)]
    backend_source: Option<String>,
}
```